src/metadata-builder/ClosureJunctionEntityMetadataBuilder.ts

```diff
@@ -110,20 +110,21 @@ export class ClosureJunctionEntityMetadataBuilder {
         }
 
         // create junction table foreign keys
+        // Note: There is only a single CASCADE because of mssql not supporting multi cascade paths
+        //       However this should not cause any problems since the both foreign keys point to the same column on the same table
```

Comment:
There is access to `this.connection.driver` here, should there be a check for `instanceof SqlServerDriver`?

Reply:
We do have access to the driver yes.. I would need to add some extra logic to the delete inside the subject executor.

I also found that the changes that I did on `TreeParent.ts` to allow deleting the parents via cascade are also causing the same problem on sql server so I'll probably need to revert that change and add custom delete logic for every time of tree entity.